2-mandatory/3-financial-times.js

```diff
@@ -15,15 +23,49 @@ function potentialHeadlines(allArticleTitles) {
 */
 function titleWithFewestWords(allArticleTitles) {
     // TODO
+    let titleLength = [];
+    for (let i = 0; i < allArticleTitles.length; i++){
+        let titleArray = allArticleTitles[i].split(" ");
+        titleLength.push(titleArray.length);
+    }
+    const minLength = Math.min(...titleLength);
+    const minIndex = titleLength.indexOf(minLength);
+    return allArticleTitles[minIndex];
 }
 
 /*
     The editor of the FT has realised that headlines which have numbers in them get more clicks!
     Implement the function below to return a new array containing all the headlines which contain a number.
     (Hint: remember that you can also loop through the characters of a string if you need to)
 */
+
+function isNumber(char) {
+  if (typeof char !== 'string') {
+    return true;
+  }
+
+  if (char.trim() === '') {
+    return false;
+  }
+
+  return !isNaN(char);
+}
+
+
 function headlinesWithNumbers(allArticleTitles) {
     // TODO
+    const arr = [];
+   for (let i = 0; i < allArticleTitles.length; i++) {
+        let article = allArticleTitles[i];
+       for (let j = 0; j < article.length; j++) {
+            let articleChar = article[j];
+          if (isNumber(articleChar)) {
+            arr.push(article);
+            break;            
+          }
+       }
+   }
+   return arr;
 }
 
```

Comment:
Why have you used break in your function?
Could you use a simple loop instead of a loop inside a loop?

Reply:
I used `break` in the function to stop searching as soon as the first July is found.
Sure, simple loop can be used.